packages/core/src/Truncate.js

```diff
@@ -1,15 +1,11 @@
 import styled from 'styled-components'
 import Text from './Text'
-import theme from './theme'
 
 const Truncate = styled(Text)`
   overflow: hidden;
   white-space: nowrap;
   text-overflow: ellipsis;
 `
-Truncate.defaultProps = {
-  theme: theme
```

Comment:
nonblocking: The reason most of these core components contain `theme` is because projects with unit tests would often have several failed tests due to undefined theme, and in some cases it was very difficult to get the theme into the test. I expect this problem to come up again if theme is removed, but it may be solvable in host apps by [mounting with theme](https://github.com/styled-components/jest-styled-components/issues/106#issuecomment-373151391)

Reply:
Definitely understand why it was added as a default prop, but unfortunately this change is must, otherwise we wouldn't see the correct theme always propagate (i.e. in class components that are not wrapped in `withTheme()`.